NeosModLoader/ModLoader.cs

```diff
@@ -22,7 +22,12 @@ internal static void LoadMods()
                 return;
             }
 
-            string modDirectory = Directory.GetParent(Process.GetCurrentProcess().MainModule.FileName).FullName + "\\nml_mods";
+            string modDirectory;
+            if (Environment.OSVersion.Platform == PlatformID.Unix)
+                modDirectory = System.IO.Path.GetDirectoryName(Assembly.GetEntryAssembly().Location)+"/nml_mods";
+            else
+                modDirectory = Directory.GetParent(Process.GetCurrentProcess().MainModule.FileName).FullName + "\\nml_mods";
+

```

Comment:
If we're going to be handling cross-platform paths, we might as well do it the right way by using [Path.Combine](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.combine?view=net-5.0) as discussed in this [stackoverflow post](https://stackoverflow.com/questions/5758633/how-should-i-handle-windows-linux-paths-in-c-sharp)

Reply:
unless you know a better way, i need to handle getting neos's path differently on windows than on Linux

Reply:
Is there a reason that this won't work for both windows and linux?
```cs
string modDirectory = Paths.Combine(Directory.GetParent(Process.GetCurrentProcess().MainModule.FileName).FullName, "nml_mods");
```

Reply:
not sure why but that fails on linux, it seems Directory.GetParent(Process.GetCurrentProcess().MainModule.FileName).FullName
is null on linux

Reply:
Weird... can you figure out where it's returning null? None of the docs for anything in that chain indicate nulls can be returned.

Alternatively, if `Path.GetDirectoryName(Assembly.GetEntryAssembly().Location)` works on Windows as well lets just ditch the Process API completely.